src/device.js

```diff
@@ -1,11 +1,33 @@
+import ADB from './adb';
+
 /**
  * Detects connected devices.
  *
- * @param {DataType} variablename - description
+ * @param {Object} [opts] - Detection options.
  * @returns {Promise}
  */
-export function detect() {
+export function detect(opts = {}) {
+	return new ADB(opts)
+		.devices()
+		.then(result => result.filter(n => { return !n.emulator; }));
 }
 
-export function install() {
+
+/**
+ * Installs the specified app to the specified device.
+ *
+ * @param {String} deviceId - The id of the device or emulator.
+ * @param {String} apkFile - The application apk file to install.
+ * @returns {Promise}
+ */
+export function install(deviceId, apkFile) {
+	if (typeof deviceId !== 'string' || !deviceId) {
```

Comment:
You don't need to validate the `deviceId` or `apkFile` here because you do it in `installApp()`.